### PR TITLE
Bag verifier supports verification of a bag in a subdirectory

### PR DESCRIPTION
**Pull request by mklander**
locate the root of the bag in the BagLocation when verifying

### CODE REVIEW

File: bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/services/VerifierTest.scala [hunk 1]
```diff
@@ -60,6 +60,40 @@ class VerifierTest
...
+    withLocalS3Bucket { bucket =>
+      withBag(
+        bucket = bucket,
+        dataFileCount = dataFileCount,
+        bagRootDirectory = Some("bag")) { bagLocation =>
+        withActorSystem { actorSystem =>
+          implicit val ec = actorSystem.dispatcher
+
+          withMaterializer { mat =>
+            implicit val _mat = mat
```
Comment (alexwlchan) [inline bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/services/VerifierTest.scala]: nit: you can collapse this into

```
withMaterializer { implicit materialiser =>
```

File: bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/Verifier.scala [hunk 2]
```diff
@@ -34,24 +36,29 @@ class Verifier(
...
     bagLocation: BagLocation
   ): Future[IngestStepResult[VerificationSummary]] = {
+
     val verificationInit =
       VerificationSummary(bagLocation = bagLocation, startTime = Instant.now)
 
     val verification = for {
+      bagRootLocation <- Future.fromTry {
+        s3BagLocator.locateBagRoot(bagLocation.objectLocation)
+      }
```
Comment (alexwlchan) [inline bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/Verifier.scala]: 🤦‍♀️

File: bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/services/VerifierTest.scala [hunk 3]
```diff
@@ -60,6 +60,40 @@ class VerifierTest
...
   }
 
+  it("passes a bag in a sub directory with correct checksums") {
+    withLocalS3Bucket { bucket =>
+      withBag(
+        bucket = bucket,
+        dataFileCount = dataFileCount,
+        bagRootDirectory = Some("bag")) { bagLocation =>
+        withActorSystem { actorSystem =>
+          implicit val ec = actorSystem.dispatcher
```
Comment (alexwlchan) [inline bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/services/VerifierTest.scala]: nit: we tend to use the Implicits.global execution context for this.

File: common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/StorageManifestService.scala [hunk 4]
```diff
@@ -23,22 +24,23 @@ import uk.ac.wellcome.platform.archive.common.storage.models.{
...
+import uk.ac.wellcome.storage.ObjectLocation
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class StorageManifestService(
   implicit
   executionContext: ExecutionContext,
   s3Client: AmazonS3
-) {
+) extends Logging {
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/StorageManifestService.scala]: Why did you add Logging? I can’t see where it’s used.

Comment (mklander) [reply common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/StorageManifestService.scala]: oh yes, think I was debugging.

**Pull request merged**